app/lib/features/chat_ng/providers/notifiers/reply_messages_notifier.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/features/chat_ng/models/reply_message_state.dart';
+import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::chat::reply_notifier');
+
+class ReplyMessageNotifier
+    extends AutoDisposeFamilyAsyncNotifier<ReplyMsgState, ReplyMsgInfo> {
+  Future<void> retryLoad() async {
+    state = const AsyncValue.loading();
+    state = await AsyncValue.guard(() => build(arg));
+  }
+
+  @override
+  Future<ReplyMsgState> build(ReplyMsgInfo arg) async {
+    // Watch messages to react to changes
+    final messages = ref.watch(
+      chatStateProvider(arg.roomId).select((value) => value.messageList),
+    );
+
+    // First try to find message in current state
+    final messageId = messages.firstWhere(
+      (eventId) => eventId == arg.originalId,
+      orElse: () => '',
+    );
+
+    if (messageId.isNotEmpty) {
+      RoomMsgId msgInfo = (arg.roomId, arg.originalId);
+      final message = ref.watch(chatRoomMessageProvider(msgInfo));
```

Comment:
I don't understand why we do anything of the above if we then just go do this ref.watch ... what is the stuff above useful for that this ref.watch wouldn't cover? Can't we move whatever is needed into the ref.watch or an adjacent provider instead?

Reply:
yes, only ```timelineStreamProvider``` is sufficient enough.